File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    private int lower_bound(int[] nums,int target,int size)
    {
        int s=0;
        int e=size-1;
        int ans=-1;
        while(s<=e)
        {
            int mid=(s+e)/2;
            if(nums[mid]==target)
            {
                ans=mid;
                e=mid-1;
            }
            else if(nums[mid]<target)
            {
                s=mid+1;
            }
            else
            {
                e=mid-1;
            }
        }
        return ans;  
    }  

    private int upper_bound(int[] nums,int target,int size)
    {
        int s=0;
        int e=size-1;
        int ans=-1;
        while(s<=e)
        {
            int mid=(s+e)/2;
            if(nums[mid]==target)
            {
                ans=mid;
                s=mid+1;;
            }
            else if(nums[mid]<target)
            {
                s=mid+1;
            }
            else
            {
                e=mid-1;
            }
        }
        return ans;  
    }
    public int[] searchRange(int[] nums, int target) {
        int res[]=new int[2];
        int size=nums.length;
        int l=lower_bound(nums,target,size);
        res[0]=l; 
        int u=upper_bound(nums,target,size); 
        res[1]=u;   
        return res;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
def lower_bound(nums,target,size):
    s=0
    e=size-1
    ans=-1
    while(s<=e):
        mid=(s+e)//2
        if(nums[mid]==target):
            ans=mid
            e=mid-1
        elif(nums[mid]<target):
            s=mid+1
        else:
            e=mid-1
    return ans 

def upper_bound(nums,target,size):
    s=0
    e=size-1
    ans=-1
    while(s<=e):
        mid=(s+e)//2
        if(nums[mid]==target):
            ans=mid
            s=mid+1
        elif(nums[mid]<target):
            s=mid+1
        else:
            e=mid-1
    return ans 

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        res=[]
        size=len(nums)
        l=lower_bound(nums,target,size)
        res.append(l)
        u=upper_bound(nums,target,size)
        res.append(u)
        return res